### PR TITLE
[DPE-4083] Run E2E integration tests on self-hosted runner

### PR DESCRIPTION
**Pull request by Batalex**

### CODE REVIEW

File: .github/workflows/ci_e2e.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ concurrency:
   cancel-in-progress: true
 
 on:
+  pull_request:
```
Comment (Batalex) [inline .github/workflows/ci_e2e.yaml:10]: **todo**: remove upon tests passing confirmation

**Comment by deusebio** [comment]
That was easy! :D

But maybe we can also use this PR to actually decide where/how to run E2E tests. We should definitely run this on scheduled runs. But I'm also thinking that maybe we should do this for pull-request and releases? Maybe tomorrow at grooming, we can briefly sync on this. It is not super important, but I'd like to settle on the pattern.

**Review by marcoppenheimer — APPROVED** [review]
LGTM!

Out of curiosity, what's the specs of `large`?

**Comment by Batalex** [comment]
> LGTM!
> 
> Out of curiosity, what's the specs of `large`?

Based on [this](https://discourse.canonical.com/t/size-changes-for-self-hosted-github-runners/3291),  8 CPUs and 32G

File: .github/workflows/ci_e2e.yaml (hunk 1 above)
Comment (Batalex) [reply .github/workflows/ci_e2e.yaml:10]: We decided to keep this running on PRs

**Review by deusebio — APPROVED** [review]
Great! Thanks!

**Pull request merged**